Manatee.Json/Internal/ReflectionExtensions.cs

```diff
@@ -87,8 +87,54 @@ public static void SetMember(this object? obj, JsonPointer pointer, object? valu
 				if (int.TryParse(segment, out var index))
 				{
 					member = ReflectionCache.GetMembers(obj.GetType(), PropertySelectionStrategy.ReadWriteOnly, false)
-						.Single(m => m.MemberInfo is PropertyInfo pi && pi.GetIndexParameters().Length == 1);
-					if (member == null) return;
+						.SingleOrDefault(m => m.MemberInfo is PropertyInfo pi && pi.GetIndexParameters().Length == 1);
+					if (member == null)
+					{
+						var jsonArray = (obj as JsonValue)?.Array ?? obj as JsonArray;

```

Comment:
i don't follow what you're trying to do here.

Reply:
Upon deserialization, it seems like the process tries to replace "$ref" references with the corresponding content.
Here I try to add functionality for resolving "$ref" references on `JsonArray` types (during deserialization).
Without `.SingleOrDefault(...)`, an exception was thrown and the subsequent `if (member == null) return;` would never hit.

Further I try to extract a `JsonArray` instance from `obj`. If it is not null, there is a JsonArray, and values could be added, not by object member, but by indices in the `JsonArray`.

The same goes for `JsonObject` down below, when the segment is not a number, but rather a member name. If the member is `null`, but `obj` is a `JsonObject`, we could proceed with traversing values in the `obj` `JsonObject`.

Reply:
Yeah, same here: we're not concerned about object references inside pure JSON content.